test-light.js

```diff
@@ -0,0 +1,439 @@
+'use strict'
+
+var test = require('tape')
+var rehype = require('rehype')
+var highlight = require('./light')
+var js = require('highlight.js/lib/languages/javascript')
+
+test('highlight()', function (t) {
+  t.equal(
+    rehype()
+      .data('settings', {fragment: true})
+      .use(highlight)
+      .processSync(
+        ['<h1>Hello World!</h1>', '', '<pre><code></code></pre>'].join('\n')
+      )
+      .toString(),
+    ['<h1>Hello World!</h1>', '', '<pre><code class="hljs"></code></pre>'].join(
+      '\n'
+    ),
+    'empty'
+  )
+
+  t.equal(
+    rehype()
+      .data('settings', {fragment: true})
+      .use(highlight, {languages: {javascript: js}})
+      .processSync(
+        [
+          '<h1>Hello World!</h1>',
+          '',
+          '<pre><code>"use strict";</code></pre>'
+        ].join('\n')
+      )
+      .toString(),
+    [
+      '<h1>Hello World!</h1>',
+      '',
+      '<pre><code class="hljs language-javascript"><span class="hljs-meta">"use strict"</span>;</code></pre>'
+    ].join('\n'),
+    'should highlight (no class)'
+  )
```

Comment:
I think these test can go in the main test file, no need to copy-paste it all!

Reply:
>> How should at least one language be enforced and which options field to use? I've used the languages field.
> 
> For testing? How do you mean?
 
If a user doesn't provide a language, there won't be any highlighting. Should it be mandatory that the user provides at least one language? Is the field "languages" a good place to put the selected languages to bundle?

> There should be some docs on this new plugin. Similar to the Browser section in lowlight (https://github.com/wooorm/lowlight#browser) perhaps?

I'll get to it at the end if it's OK with you.

Reply:
> 
> 
> I think these test can go in the main test file, no need to copy-paste it all!

I suppose this is related to my question:
Should I just duplicate the tests? I didn't have time to understand them all. I'll move them to the main test.js once they work. I'd prefer to go one by one from a clean slate in a different file for starters.

Reply:
yes, for the tests, indeed!

Reply:
> If a user doesn't provide a language, there won't be any highlighting. Should it be mandatory that the user provides at least one language? Is the field "languages" a good place to put the selected languages to bundle?

I think that should be clearly documented. I don’t think we need to do extra work to go into the `languages` object to see if there’s at least one key or not 🤷‍♂️

Reply:
> > If a user doesn't provide a language, there won't be any highlighting. Should it be mandatory that the user provides at least one language? Is the field "languages" a good place to put the selected languages to bundle?
> 
> I think that should be clearly documented. I don’t think we need to do extra work to go into the `languages` object to see if there’s at least one key or not 🤷‍♂️

"no extra work " seems fine with me :grin: